src/messaging/ExchangeMgr.cpp

```diff
@@ -65,20 +67,27 @@ ExchangeManager::ExchangeManager()
 
 CHIP_ERROR ExchangeManager::Init(SecureSessionMgr * sessionMgr)
 {
+    ExchangeContext * ec = mContextPool;
+
     if (mState != State::kState_NotInitialized)
         return CHIP_ERROR_INCORRECT_STATE;
 
     mSessionMgr = sessionMgr;
 
     mNextExchangeId = GetRandU16();
 
+    for (int i = 0; i < CHIP_CONFIG_MAX_EXCHANGE_CONTEXTS; i++, ec++)
+        ec = nullptr;
+
     mContextsInUse = 0;
 
     memset(UMHandlerPool, 0, sizeof(UMHandlerPool));
     OnExchangeContextChanged = nullptr;
 
     sessionMgr->SetDelegate(this);
 
+    mReliableMessageMgr.Init(sessionMgr->SystemLayer(), sessionMgr, mContextPool);
```

Comment:
This just wants to give `mReliableMessageMgr` an alias to our context pool, right?  Why not do that directly?  Have it store a reference in its constructor, and then we don't need to do anything to pass the value around in `Init`.  It will also then be able to know the size of the pool directly, use ranged for loops, and so forth...

Reply:
That is the exactly the method I mentioned in my previous comments, I am not quite familiar how std::array is implemented, if it could be moved by std in some cases like std::vector, will this alias in ReliableMessageMgr guaranteed to be still valid in this scenario?

Reply:
`std::array` can `swap`, but that doesn't invalidate the memory, just changes what objects are stored in it.  And again, this would be a reference to the _object_, not to its memory.

If you are really worried about `std::array`, you could also use a C-style array as now, and store a reference to _that_ here.  Really, the right thing to do is to typedef the pool type, store a reference to the typedef type here, and then we can make the pool type be whatever makes sense.

Reply:
Acked, changed to using reference